modules/GKB/EnsEMBLMartUtils.pm

```diff
@@ -196,22 +209,73 @@ sub get_xml_query {
     return <<XML;
 <?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE Query>
-<Query  virtualSchemaName = "$virtual_schema" formatter = "TSV" header = "0" uniqueRows = "0" count = "" completionStamp = "1">	
+<Query  virtualSchemaName = "$virtual_schema" formatter = "TSV" header = "0" uniqueRows = "0" count = "" completionStamp = "1">
     <Dataset name = "$dataset" interface = "default" >
 $attribute_tags
     </Dataset>
 </Query>
 XML
 }
 
+sub get_wget_query_for_identifier {
+    my $logger = get_logger(__PACKAGE__);
+
+    my $species_abbreviation = shift // $logger->logconfess("Need species abbreviation");
+    my $identifier = shift // $logger->logconfess("Need identifier to query");
+
+    return "wget -q -O - '" . get_mart_info_for_species($species_abbreviation)->{'url'} . "?query=" .
+        get_xml_query_for_identifier($species_abbreviation, $identifier) . "'";
+}
+
+sub get_xml_query_for_identifier {
+    my $logger = get_logger(__PACKAGE__);
+
+    my $species_abbreviation = shift // $logger->logconfess("Need species abbreviation");
+    my $identifier = shift // $logger->logconfess("Need identifier to query");
+
+    my $mart_info = get_mart_info_for_species($species_abbreviation);
+
+    my $dataset = $mart_info->{'dataset'};
+    my $virtual_schema = $mart_info->{'virtual_schema'};
+
+    $dataset // confess "No dataset defined\n";
+    $virtual_schema // confess "No virtual schema defined\n";
+
+    return <<XML;
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE Query>
+<Query  virtualSchemaName = "$virtual_schema" formatter = "TSV" header = "0" uniqueRows = "0" count = "" completionStamp = "1">
+    <Dataset name = "$dataset" interface = "default" >
+        <Attribute name = "ensembl_gene_id" />
+        <Attribute name = "ensembl_transcript_id" />
+        <Attribute name = "ensembl_peptide_id" />
+        <Attribute name = "$identifier" />
+    </Dataset>
+</Query>
+XML
+}
+
+sub get_species_mart_name {
+    my $logger = get_logger(__PACKAGE__);
+
+    my $species_abbreviation = shift // $logger->logconfess("Need species abbreviation");
+
+    my $species_name = $species_info{$species_abbreviation}->{'name'}->[0];
+    if ($species_name =~ /^(\w)\w+ (\w+)$/) {

```

Comment:
I'm not sure about the regex... will it work if there is more than one "word"? For example: "Human immunodeficiency virus 1" (DBID: 164493)? Or "Neisseria meningitidis serogroup B" (DBID: 180800)? And "Influenzavirus A" (DBID: 168239) will be turned in to "IA", by this regex, which doesn't seem to be a useful abbreviation (at least not to me).

(I did some testing: regexr.com/4985h)

It might not matter, if this code is only expected to deal with species names that are exactly two words and that are expected to match with species names/codes in the mart.

Reply:
`$species_name` will get its value from the Config_Species.pm file, so that currently only has species with two words.  If it doesn't match the two word pattern, though, it will trigger the else and log an error since its not obvious how to construct a species name for EnsEMBL Mart if there aren't exactly two words for the species.  

I'll update it, though, to throw an error instead of just logging and returning undefined.